azure-keyvault/src/main/java/com/microsoft/azure/keyvault/authentication/KeyVaultCredentials.java

```diff
@@ -96,6 +103,23 @@ public Response intercept(Chain chain) throws IOException {
             Map<String, String> challengeMap) throws IOException {
 
         Boolean supportsPop = supportsMessageProtection(originalRequest.url().toString(), challengeMap);
+
+        // if the service supports pop and a clientEncryptionKey has not been generated yet, generate
+        // the key that will be used for encryption on this an all subsequent protected requests
+        if (supportsPop && this.clientEncryptionKey == null)
+        {
+            try {
+                final KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA");
+
+                generator.initialize(2048);
+                
+                this.clientEncryptionKey = JsonWebKey.fromRSA(generator.generateKeyPair()).withKid(UUID.randomUUID().toString());   
+                
+            } catch (NoSuchAlgorithmException e) {
+                // Unexpected. Should never be thrown. 

```

Comment:
Do we really want to just ignore this if it gets thrown? I realize it's not expected, but we probably shouldn't suppress it does happen.